clusterloader2/pkg/flags/flags.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flags
+
+import (
+	"flag"
+	"os"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+)
+
+const (
+	noInt    = -12345
+	noString = "xxxNoStringxxx"
+)
+
+func init() {
+	pflag.CommandLine = pflag.NewFlagSet(os.Args[0], pflag.ContinueOnError)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+}
+
+type flagEnv interface {
+	ParseEnv()
+}
+
+type stringFlagEnv struct {
+	v            *string
+	envVariable  string
+	defaultValue string
+}
+
+// ParseEnv assigns env variable value if needed.
+func (s *stringFlagEnv) ParseEnv() {
+	if *s.v == noString {
+		if s.envVariable != "" {
+			if val, ok := os.LookupEnv(s.envVariable); ok {
+				*s.v = val
+				return
+			}
+		}
+		*s.v = s.defaultValue
+	}
+}
+
+type intFlagEnv struct {
+	v            *int
+	envVariable  string
+	defaultValue int
+}
+
+// ParseEnv assigns env variable value if needed.
+func (i *intFlagEnv) ParseEnv() {
+	if *i.v == noInt {
+		if i.envVariable != "" {
+			if val, ok := os.LookupEnv(i.envVariable); ok {
+				iVal, err := strconv.Atoi(val)
+				if err != nil {
+					*i.v = iVal
+					return
+				}
+				glog.Errorf("Parsing env variable %s failed. Using default %d", i.envVariable, i.defaultValue)
```

Comment:
I would say that we should fail the test if we are not able to parse the flag.

Reply:
Done.